support-frontend/assets/helpers/internationalisation/helpers.ts

```diff
@@ -0,0 +1,411 @@
+import * as cookie from 'helpers/storage/cookie';
+import type { Option } from 'helpers/types/option';
+import { getQueryParameter } from 'helpers/urls/url';
+import type {
+	AuState,
+	CaState,
+	IsoCountry,
+	StateProvince,
+	UsState,
+} from './country';
+import {
+	auStates,
+	caStates,
+	countries,
+	isoCountries,
+	isoCountrySet,
+	usStates,
+} from './country';
+import {
+	AUDCountries,
+	Canada,
+	countryGroups,
+	EURCountries,
+	GBPCountries,
+	International,
+	NZDCountries,
+	UnitedStates,
+} from './countryGroup';
+import type {
+	CountryGroup,
+	CountryGroupId,
+	CountryGroupName,
+} from './countryGroup';
+
+type TargetCountryGroups =
+	| typeof International
+	| typeof EURCountries
+	| typeof NZDCountries
+	| typeof GBPCountries
+	| typeof AUDCountries;
+
+class CountryHelper {
```

Comment:
Fully expected this question. In order to solve the dependency cycle I found that the best way was to take the helper functions out of the `country` and `countryGroup` files. The issue with that was that a lot of them shared the same names like `detect` and `fromString`, so rather than make the names of those functions long winded and more specific or separate them into separate files which felt like adding unnecessary untidiness, I thought keep them in the same file and namespace them. I played around with putting them in separate objects but a lot of those functions reference each other so I thought the nice, tidy and modern approach of using classes with static methods was best as it'll handle scoping and namespacing less messily and gives us `Country` and `CountryGroup` classes similar to the native `Math` and `Object` classes in JS